.github/workflows/pkgci_shark_ai.yml

```diff
@@ -25,17 +25,25 @@ jobs:
     strategy:
       matrix:
         version: [3.11]
+        test_device: ["cpu", "gfx90a", "gfx942"] # temporarily not running on "gfx942" due to contention with other ci tasks
       fail-fast: false
-    runs-on: azure-cpubuilder-linux-scale
-    # runs-on: ubuntu-latest # everything else works but this throws an "out of resources" during model loading
-    # TODO: make a copy of this that runs on standard runners with tiny llama instead of a 8b model
+    runs-on: >-
+      ${{
+        matrix.test_device == 'cpu' && 'azure-cpubuilder-linux-scale' ||
+        matrix.test_device == 'gfx942' && 'linux-mi300-gpu-1' ||
+        matrix.test_device == 'gfx90a' && 'nodai-amdgpu-mi250-x86-64'
+      }}
```

Comment:
How about an explicit matrix instead of putting this logic in?

For example: https://github.com/iree-org/iree/blob/5f7b471f88e9c8e145637bc1d2f7a4c1d8b4462e/.github/workflows/pkgci_regression_test.yml#L29-L56

```yml
strategy:
  matrix:
    - test_device: cpu
      runs-on: azure-cpubuilder-linux-scale
    - test_device: gfx90a
      runs-on: nodai-amdgpu-mi250-x86-64
```

Docs: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow

Reply:
thanks! that looks MUCH cleaner!